src/Core/Settings.h

```diff
@@ -601,6 +601,10 @@ static constexpr UInt64 operator""_GiB(unsigned long long value)
     M(Bool, throw_if_no_data_to_insert, true, "Enables or disables empty INSERTs, enabled by default", 0) \
     M(Bool, compatibility_ignore_auto_increment_in_create_table, false, "Ignore AUTO_INCREMENT keyword in column declaration if true, otherwise return error. It simplifies migration from MySQL", 0) \
     M(Bool, multiple_joins_try_to_keep_original_names, false, "Do not add aliases to top level expression list on multiple joins rewrite", 0) \
+    M(Bool, allow_grace_hash_join, false, "Allow selecting grace hash join in 'auto' join algorithm", 0) \
```

Comment:
In current `master` branch `join_algorithm` supports multiple values (if future clickhouse should be able to choose best algrithm from provided list, now it just uses first avaliable, because some of them works only with specific storages like RocksDB or Dictionary). So, we can use join_algorithm = 'auto,grace_hash` instead of this additional setting